plans/extend_ca_cert.pp

```diff
@@ -2,6 +2,7 @@
   TargetSpec $targets,
   Optional[TargetSpec] $compile_masters = undef,
   $ssldir                               = '/etc/puppetlabs/puppet/ssl',
+  $regen_primary_cert                   = false,
```

Comment:
It seems as though we should default this to `true` and make it clear we can disable it.

Reply:
Are you ok with defaulting to `false` with the new error and warning messages?  I also added the param to the reference.